src/async_client.rs

```diff
@@ -275,154 +260,38 @@ impl AsyncClient {
         &self.homeserver
     }
 
-    /// Calculate the room name from a `RoomId`, returning a string.
+    /// Add `EventEmitter` to `AsyncClient`.
+    ///
+    /// The methods of `EventEmitter` are called when the respective `RoomEvents` occur.
+    pub async fn add_event_emitter(
+        &mut self,
+        emitter: Arc<tokio::sync::Mutex<Box<dyn EventEmitter>>>,
+    ) {
+        self.base_client.write().await.event_emitter = Some(emitter);
```

Comment:
The base client needs to be usable on it's own, so it should have a method to add the event emitter as well.